OpenRA.Mods.Common/Traits/World/EditorResourceLayer.cs

```diff
@@ -58,6 +60,27 @@ public void WorldLoaded(World w, WorldRenderer wr)
 
 			foreach (var cell in Map.AllCells)
 				UpdateCell(cell);
+
+			// Build the sprite layer dictionary for rendering resources
+			// All resources that have the same palette must also share a sheet and blend mode
```

Comment:
`EditorResourceLayer` duplicates a bit of `ResourecLayer`, but there is significantly more code in there which it doesn't (and shouldn't) use.  I made the best of two bad choices by splitting it off into a different trait with some duplication vs shoehorning more complexity into `ResourceLayer` by forcing it to also work with the editor.  Unless you have a specific suggestion there's not much I can do here.